utilities/file_reader.py

```diff
@@ -0,0 +1,3 @@
+def generate(file):
+    for line in reversed(list(open(file))):
+        yield line
```

Comment:
It is good that the logs are being displayed `most recent first`. Try to group them by the `timestamp` that precedes every log.

Reply:
The most recent logs are displayed first. This is what was requested of me @rajeman.

Reply:
Furthermore, not every line in the log file is preceded by a timestamp.

Reply:
You can look through the file again and verify every log is preceded by a timestamp.  If you need clarification, kindly reach out to me.

Reply:
@rajeman from the following screenshot, you can see that the some lines do not contain the preceding timestamp.
![image](https://user-images.githubusercontent.com/41931013/62520413-36e17200-b836-11e9-8a3b-cf56e2da90cb.png)